File: itx-go/itx-contract/repo/factory/factory.go
package factory

import (
	"github.com/chehsunliu/itx/itx-go/itx-contract/repo/post"
	"github.com/chehsunliu/itx/itx-go/itx-contract/repo/subscription"
	"github.com/chehsunliu/itx/itx-go/itx-contract/repo/user"
)

type RepoFactory interface {
	CreatePostRepo() post.Repo
	CreateUserRepo() user.Repo
	CreateSubscriptionRepo() subscription.Repo
}

File: itx-rs/itx-contract/src/repo/factory.rs
use std::sync::Arc;

use crate::repo::post::PostRepo;
use crate::repo::user::UserRepo;

pub trait RepoFactory: Send + Sync {
    fn create_post_repo(&self) -> Arc<dyn PostRepo>;
    fn create_user_repo(&self) -> Arc<dyn UserRepo>;
}
